earth_enterprise/rpms/pre-uninstall_server.sh

```diff
@@ -0,0 +1,159 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set +x
+
+#------------------------------------------------------------------------------
+# from common.sh:
+# versions and user names
+GEE="Google Earth Enterprise"
+
+# directory locations
+BININSTALLROOTDIR="/etc/init.d"
+BASEINSTALLDIR_OPT="/opt/google"
+BASEINSTALLDIR_ETC="/etc/opt/google"
+BASEINSTALLDIR_VAR="/var/opt/google"
+
+check_server_processes_running()
+{
+  printf "\nChecking geserver services:\n"
+  local retval=1
+
+  # i) Check if postgres is running
+  local post_master_running=$( ps -ef | grep postgres | grep -v grep )
+  local post_master_running_str="false"
+
+  # ii) Check if gehttpd is running
+  local gehttpd_running=$( ps -ef | grep gehttpd | grep -v grep )
+  local gehttpd_running_str="false"
+
+  # iii) Check if wsgi is running
+  local wsgi_running=$( ps -ef | grep wsgi:ge_ | grep -v grep )
+  local wsgi_running_str="false"
+
+  if [ -n "$post_master_running" ]; then
+    retval=0
+    post_master_running_str="true"
+  fi
+  echo "postgres service: $post_master_running_str"
+
+  if [ -n "$gehttpd_running" ]; then
+    retval=0
+    gehttpd_running_str="true"
+  fi
+  echo "gehttpd service: $gehttpd_running_str"
+
+  if [ -n "$wsgi_running" ]; then
+    retval=0
+    wsgi_running_str="true"
+  fi
+  echo "wsgi service: $gehttpd_running_str"
+
+  return $retval
+}
+
+#------------------------------------------------------------------------------
+# script arguments
+DELETE_GROUP=true
+
+# user names
+GEAPACHEUSER_NAME=""
+GEPGUSER_NAME=""
+GEGROUP_NAME=""
+GEAPACHEUSER_EXISTS=""
+GEPGUSER_EXISTS=""
+GEGROUP_EXISTS=""
+
+PUBLISH_ROOT_CONFIG_PATH="$BASEINSTALLDIR_OPT/gehttpd/conf.d"
+
+# additional variables
+HAS_FUSION=false
+SEARCH_SPACE_PATH=""
+STREAM_SPACE_PATH=""
+
+main_preuninstall()
+{
+  # check to see if GE Server processes are running
+  if check_server_processes_running; then
+    #STOP PROCESSES
+  fi
+
+  # Determine if fusion is installed
+  if [ -f "$BININSTALLROOTDIR/gefusion" ]; then
+    HAS_FUSION=true
+  else
+    HAS_FUSION=false
+  fi
+
+  # get the GE user names
+  get_user_names
+
+  # check if the group can be safely deleted
+  check_group_delete
+
+  # find the publish root
+  get_publish_roots
+}
+
+get_user_names()
+{
+  GEAPACHEUSER_NAME=`cat $BININSTALLROOTDIR/gevars.sh | grep GEAPACHEUSER | cut  -d'=' -f2`
+  GEPGUSER_NAME=`cat $BININSTALLROOTDIR/gevars.sh | grep GEPGUSER | cut  -d'=' -f2`
+  GEGROUP_NAME=`cat $BININSTALLROOTDIR/gevars.sh | grep GEGROUP | cut  -d'=' -f2`
+
+  # Make sure the users and group exist
+  GEAPACHEUSER_EXISTS=$(getent passwd $GEAPACHEUSER_NAME)
+  GEPGUSER_EXISTS=$(getent passwd $GEPGUSER_NAME)
+  GEGROUP_EXISTS=$(getent group $GEGROUP_NAME)
+}
+
+check_group_delete()
```

Comment:
We should try to either separate Fusion and Server groups, or move this to the `common` package.